### PR TITLE
Expose type and sexp

### PR DESCRIPTION
**Pull request by acjay**
I _think_ this should do it.

### CODE REVIEW

File: statsd-client-async.opam
Comment (brendanlong) [inline statsd-client-async.opam]: Why did we need to change this?

File: statsd-client-async/jbuild
Comment (brendanlong) [inline statsd-client-async/jbuild]: It would be nice to alphabetize library names where possible, but it's not a huge deal.

File: statsd-client/dogstatsd.ml
Comment (brendanlong) [inline statsd-client/dogstatsd.ml]: Do we actually need both of these opens? I though Jane Street stuff usually didn't need `.Std` opened anymore

File: statsd-client/dogstatsd.ml
Comment (brendanlong) [inline statsd-client/dogstatsd.ml]: It's generally nice to put the `[@@deriving ...]` on the next line for multi-line types

**Review by brendanlong — APPROVED** [review]
Looks good. I left some nitpicks in the comments but this is good to merge if you want.

File: statsd-client-async.opam
Comment (acjay) [reply statsd-client-async.opam]: Ah, this is the version of Opam? I was thinking it was the version of _this_ package. Will roll that back, if so.

File: statsd-client-async/jbuild
Comment (acjay) [reply statsd-client-async/jbuild]: Good idea. An old coworker had the opinion that almost everything should be alphabetized, unless it has an extremely obvious or semantically important alternative ordering. I thought that was a bit radical at first, but after a while, it really grew on me.

File: statsd-client/dogstatsd.ml
Comment (acjay) [reply statsd-client/dogstatsd.ml]: Hmm, dropping `.Std`  appears to break the build on my machine.

File: statsd-client-async.opam
Comment (brendanlong) [reply statsd-client-async.opam]: Yeah this is the opam version. Now that I think about it, I don't think 1.3 exists (the latest two versions are 1.2 and 2.0)

File: statsd-client/dogstatsd.ml
Comment (brendanlong) [reply statsd-client/dogstatsd.ml]: Huh ok ¯\_(ツ)_/¯

**Review by brendanlong — APPROVED** [review]
Looks good!

**Comment by acjay** [comment]
@brendanlong Refactored a bit to cut out the indirection and duplication that was confusing me.

**Comment by acjay** [comment]
In testing this with local Godzilla, it works, but references to `Statsd_client_async__.Dogstatsd.Metric` are now `Statsd_client_async.Dogstatsd.Metric`, which is a breaking change. We'll have to merge this in quick succession with landing the change to Godzilla.

File: statsd-client/dogstatsd.ml
Comment (brendanlong) [inline statsd-client/dogstatsd.ml]: You could just replace this with `module T = Dogstatsd_types.Types`

File: statsd-client/dogstatsd_types.ml
Comment (brendanlong) [inline statsd-client/dogstatsd_types.ml]: The convention would be to name this file `Dogstatsd_intf` and the module inside it `S`, so `Dogstatsd_types.Types` would be come `Dogstatsd_intf.S`.

**Review by brendanlong — APPROVED** [review]
I think this looks good, although it's hard for me to review the "DRY" commit. What we should probably do at some point is break out a bunch of modules into their own files, but I'd rather not make this pull request any bigger.

File: statsd-client/dogstatsd.ml
Comment (brendanlong) [reply statsd-client/dogstatsd.ml]: Also (following OCaml conventions), this module should probably be named `S`, not `T`, since interfaces tend to be named `S` and `T` tends to used for concrete types.

Comment (acjay) [reply statsd-client/dogstatsd.ml]: Ah, interesting. Not having made it through RWO all the way yet, do you happen to know whether it will cover most of these conventsions, or whether there's a good source?

The DRY diff isn't pretty, but it's literally just the body of `T` in both files factored out. So this is the same `T` as before. Combining this with your comment below, would the idea be to do `module T = Dogstatsd_intf.S`, or should it have more appropriately been called `S` prior to this PR?

Comment (brendanlong) [reply statsd-client/dogstatsd.ml]: It should probably have been named `S` in the first place.

Unfortunately there's not a great source for the style or conventions. We should probably write one..

**Review by brendanlong — APPROVED** [review]
Looks good :)

**Comment by brendanlong** [comment]
You have permissions to merge your own PR's once it's approved right?

**Comment by acjay** [comment]
Yep! Gotta prep Godzilla first so I don't break that build.

**Comment by brendanlong** [comment]
Doesn't godzilla pin a particular version? If not we should do that.

**Comment by acjay** [comment]
I don't believe it does, but I'm also not 100% on this. See https://work.pggd.co/D11610 for my slightly informed attempt to fix this.

**Comment by acjay** [comment]
Alright, Godzilla is ready. Merging!

**Pull request merged**
